### PR TITLE
Game Event Running

### PR DESCRIPTION
**Pull request by noahklowden**
This pull requests all schedule game events. The game can also be started by setting "gameStarted" to true. This should trigger the game start in the admin player, which updates the setting for every other game object for every other user.

### CODE REVIEW

File: appGeo/src/app/game/game.component.ts [hunk 1]
```diff
@@ -89,6 +102,31 @@ export class Game implements OnInit {
...
+  updateEndTime(data: object) {
+    this.endTime = data["value"]
+    console.log("updated end time" + JSON.stringify(this.endTime))
+  }
+
+  updateStatus(data: object) {
+    this.gameActive = data["value"]
+    console.log("updated status" + JSON.stringify(this.gameActive))
+  }
+
```
Comment (gypark23) [inline appGeo/src/app/game/game.component.ts:129]: Are these functions supposed to do something? Are we putting listeners just to console.log?

File: appGeo/src/app/game/game.component.ts [hunk 2]
```diff
@@ -73,6 +80,12 @@ export class Game implements OnInit {
         this.players.forEach((player: Player, key: number) => {
           databaseEventListener(player.getDatabasePath(), this.updatePlayerDatabase.bind(this));
         });
+        databaseEventListener(GAMERULE_PATH, this.updateGameRulesDatabase.bind(this));
+        databaseEventListener(MAP_PATH, this.updateMapDatabase.bind(this));
+        databaseEventListener(START_TIME_PATH, this.updateStartTime.bind(this));
+        databaseEventListener(END_TIME_PATH, this.updateEndTime.bind(this));
+        databaseEventListener(STATUS_PATH, this.updateStatus.bind(this));
+
```
Comment (gypark23) [inline appGeo/src/app/game/game.component.ts:88]: Not sure why we are adding listeners in the forEach loop for each player when seemingly these listeners aren't even player specific

**Review by gypark23 — CHANGES_REQUESTED** [review]
Don't need explicit changes but additional clarifications would be nice, especially since it is hard to debug your code given we actually need to simulate the game to find bugs

File: appGeo/src/app/game/game.component.ts (hunk 1 above)
Comment (noahklowden) [reply appGeo/src/app/game/game.component.ts:129]: All of these functions are used to update all the game classes for each user to match. Notice how they all are setting values

File: appGeo/src/app/game/game.component.ts (hunk 2 above)
Comment (noahklowden) [reply appGeo/src/app/game/game.component.ts:88]: These aren't in the forEach loop? These are all after.

Comment (gypark23) [reply appGeo/src/app/game/game.component.ts:88]: my bad LOL I must be out of my mind right now

**Review by gypark23 — APPROVED** [review]

**Pull request merged**